### ISSUE TITLE
Response headers keys being converted to lowercase

### ISSUE THREAD

**Issue by bharling** [encode/apistar#358]
APIStar seems to be lowercasing Response headers, not sure why but it seems to break some client side things expecting title-cased headers eg:

`def login(email:str, password:str):
    data = {"status":"success"}
    return Response(data, status=200, headers=(('Authorization', 'sfksdjfdjk'),))`

`$ curl -v -d '{"email":"value1", "password":"value2"}' -H "Content-Type: application/json" -X POST http://192.168.99.100:8000/auth/login
Note: Unnecessary use of -X or --request, POST is already inferred.
* STATE: INIT => CONNECT handle 0x6000573e0; line 1407 (connection #-5000)
* Added connection 0. The cache now contains 1 members
*   Trying 192.168.99.100...
* TCP_NODELAY set
* STATE: CONNECT => WAITCONNECT handle 0x6000573e0; line 1460 (connection #0)
* Connected to 192.168.99.100 (192.168.99.100) port 8000 (#0)
* STATE: WAITCONNECT => SENDPROTOCONNECT handle 0x6000573e0; line 1567 (connection #0)
* Marked for [keep alive]: HTTP default
* STATE: SENDPROTOCONNECT => DO handle 0x6000573e0; line 1585 (connection #0)
> POST /auth/login HTTP/1.1
> Host: 192.168.99.100:8000
> User-Agent: curl/7.51.0
> Accept: */*
> Content-Type: application/json
> Content-Length: 39
>
* upload completely sent off: 39 out of 39 bytes
* STATE: DO => DO_DONE handle 0x6000573e0; line 1664 (connection #0)
* STATE: DO_DONE => WAITPERFORM handle 0x6000573e0; line 1791 (connection #0)
* STATE: WAITPERFORM => PERFORM handle 0x6000573e0; line 1801 (connection #0)
* HTTP 1.0, assume close after body
* Marked for [closure]: HTTP/1.0 close after body
< HTTP/1.0 200 OK
< authorization: sfksdjfdjk
< content-type: application/json
< Connection: close
< Server: Werkzeug/0.12.2 Python/3.6.1
< Date: Sun, 12 Nov 2017 20:41:17 GMT
<
* nread <= 0, server closed connection, bailing
* STATE: PERFORM => DONE handle 0x6000573e0; line 1965 (connection #0)
* multi_done
* Curl_http_done: called premature == 0
* Closing connection 0
* The cache now contains 0 members
* Expire cleared
{"status": "success"}`

Note both the 'content-type' and 'authorization' response headers are lowercase.

**Comment by seanhead** [comment]
https://tools.ietf.org/html/rfc7230#section-3.2

>  Each header field consists of a case-insensitive field name followed by a colon (":"), optional leading whitespace, the field value, and optional trailing whitespace.

**Comment by funkybob** [comment]
So in short, if your client has issues with this, it is at fault for not complying with the standard. I'm guessing APIstar works this ways as it's simpler to lower-case all header names than to deal with ci-compare and all that.

**Comment by bharling** [comment]
Yeah I had a refresh of the standard after posting, just thought it was worth mentioning though as I'd have expected the case to be preserved, and I've not seen this with other frameworks.

**Comment by bharling** [comment]
Feel free to close though as it's clearly not a serious issue

**Comment by funkybob** [comment]
It's probably worth noting in the docs somewhere, since clearly people are going to get caught by it.

I learned this same lesson when writing a serverless toolkit...
